src/status_im/components/drawer/view.cljs

```diff
@@ -49,26 +43,23 @@
 
 (defview name-input [placeholder]
   [account [:get-current-account]
-   new-name [:get-in [:profile-edit :name]]]
-  (let [current-name (:name account)]
-    [view {:style st/name-input-wrapper}
-     [text-input
-      {:placeholder    placeholder
-       :style          (st/name-input-text (s/valid? ::profile.db/name (or new-name current-name)))
-       :font           :medium
-       :default-value  (or new-name current-name)
-       :on-change-text #(rf/dispatch [:set-in [:profile-edit :name] %])
-       :on-end-editing #(do
-                          (rf/dispatch [:set-in [:profile-edit :name] nil])
-                          (when (s/valid? ::profile.db/name new-name)
-                            (rf/dispatch [:account-update {:name (utils/clean-text new-name)}])))}]]))
+   name-text  (r/atom nil)]
+  [view st/name-input-wrapper
+   [text-input
+    {:placeholder    placeholder
+     :style          (st/name-input-text (s/valid? ::profile.db/name @name-text))
+     :font           :medium
+     :default-value  (or @name-text (:name account))
+     :on-change-text #(reset! name-text %)
+     :on-end-editing #(if (s/valid? ::profile.db/name @name-text)
```

Comment:
I agree here I didn't really change anything regarding validation it was done like this before